src/plugins/intel_gna/src/preprocessing.cpp

```diff
@@ -38,3 +41,267 @@ void GNAPluginNS::ConvertToInt16(int16_t *ptr_dst,
         ptr_dst[i] = ConvertFloatToInt16(ptr_src[i]*scale_factor);
     }
 }
+
+#ifdef HAVE_AVX2
+void GNAPluginNS::ConvertMatrixFp32ToInt16(int16_t* ptr_dst,
```

Comment:
You need to move this to a separate file and compile only this file with AVX2 support. 
In the another file which uses this function, you need to check in runtime whether AVX2 is here and call the method. 

It's not allowed to mix up AVX2 and plain code in the same file, because when you compiled everything with AVX2, compiler can insert vector instructions even in scalar code. And later you code will fail with illegal instruction on SSE-only processors.